src/sentry/seer/fetch_issues/fetch_issues_given_exception_type.py

```diff
@@ -0,0 +1,102 @@
+import logging
+from datetime import UTC, datetime, timedelta
+
+from sentry.integrations.models.repository_project_path_config import RepositoryProjectPathConfig
+from sentry.models.group import Group
+from sentry.models.repository import Repository
+
+MAX_NUM_ISSUES_DEFAULT = 5
+"""
+The maximum number of potentially related issues to return.
+"""
+
+NUM_DAYS_AGO = 90
+"""
+The number of previous days from now to find issues and events.
+This number is global so that fetching issues and events is consistent.
+"""
+
+logger = logging.getLogger(__name__)
+
+
+def get_issues_related_to_exception_type(
+    *,
+    organization_id: int,
+    provider: str,
+    external_id: str,
+    exception_type: str,
+    max_num_issues: int = MAX_NUM_ISSUES_DEFAULT,
+    num_days_ago: int = NUM_DAYS_AGO,
+    run_id: int | None = None,
+) -> list[int]:
+    """
+    Fetches issue ids with the given exception types.
+
+    Args:
+        organization_id: The organization id.
+        provider: The provider of the repository.
+        external_id: The external id of the repository.
+        exception_type: The exception type to fetch issues for.
+        max_num_issues: The maximum number of issues to fetch.
+        num_days_ago: The number of days ago to fetch issues for.
+        run_id: The run id.
+
+    Returns:
+        A dictionary of exception types and their corresponding issue ids.
+    """
+
+    try:
+        repo = Repository.objects.get(
+            organization_id=organization_id, provider=provider, external_id=external_id
+        )
+    except Repository.DoesNotExist:
+        logger.exception(
+            "Repo doesn't exist",
+            extra={
+                "organization_id": organization_id,
+                "provider": provider,
+                "external_id": external_id,
+                "run_id": run_id,
+            },
+        )
+        return []
+
+    repo_id = repo.id
+
+    try:
+        repo_project_path_configs = RepositoryProjectPathConfig.objects.filter(
+            organization_id=organization_id,
+            repository_id=repo_id,
+        )
+        if repo_project_path_configs.count() == 0:
+            raise RepositoryProjectPathConfig.DoesNotExist
+    except RepositoryProjectPathConfig.DoesNotExist:
+        logger.exception(
+            "Repo project path config doesn't exist",
+            extra={
+                "organization_id": organization_id,
+                "provider": provider,
+                "external_id": external_id,
+            },
+        )
+        return []
```

Comment:
same comment re specific error here. not sure what the right message is. but good to be specific for le agent to intelligently iterate